src/settings/tabcpp.cpp

```diff
@@ -0,0 +1,156 @@
+    mpComboStandart->setCurrentIndex(0);
+    for(int i = 0; i < mStandartList.size(); ++i)
+    {
+        auto &item = mStandartList[i];
+        if (item == mStandartCurrernt)
+        {
+            mpComboStandart->setCurrentIndex(i);
+            break;
+        }
+    }
+    connect(mpComboStandart, SIGNAL(currentIndexChanged(const QString&)),
+                            SLOT(onChangeStandart(const QString&)));
+
+    //  path to compilers
+    QVBoxLayout *compilersLayout = new QVBoxLayout;//();
+    compilersLayout->addStretch(mBasicStretch);
+    QLabel *labelCompiler = new QLabel(tr("Compilers:"));
+    mpCompilersList = new QListWidget(this);
+    mpCompilersList->addItems(mCompilersList);
+    mpCompilersList->setSelectionMode(QAbstractItemView::SingleSelection);
+
+    compilersLayout->addWidget(labelCompiler);
+    compilersLayout->addWidget(mpCompilersList);
+
+    mpAddCompiler = new QPushButton("Add ...", this);
+    mpAddCompiler->setMaximumWidth(mCompilerWidth);
+    mpAddCompiler->setMinimumWidth(mCompilerWidth);
+    compilersLayout->addWidget(mpAddCompiler);
+    connect(mpAddCompiler, &QPushButton::clicked, this, &TabCpp::onAddCompilerClicked);
+    connect(mpCompilersList, &QListWidget::itemDoubleClicked, this, &TabCpp::onCompilerListDoubleClicked);
+
+
+
+    mainLayout->addLayout(compilersLayout);
+    mainLayout->addSpacing(mColumnSpasing);
+
+    mainLayout->addStretch(mStretchFactorMain);
+    setLayout(mainLayout);
+}
+
+const QString &TabCpp::getStandartCurrernt() const
+{
+    return mStandartCurrernt;
+}
+
+const QString &TabCpp::getStandartNew() const
+{
+    return mStandartNew;
+}
+
+void TabCpp::setStandartCurrernt(const QString & newItem)
+{
+    mStandartCurrernt = newItem;
+}
+
+void TabCpp::onChangeStandart(const QString & newItem)
+{
+    mStandartNew = newItem;
+}
+
+void TabCpp::onAddCompilerClicked()
+{
+    QString text = QInputDialog::getText(this, "Create new path", "Type path to compiler, please ...");
+    if (text.size())
+    {
+        mpCompilersList->addItem(text);     //  add item to widget
+        mCompilersList << text;             //  add item ti list
+        QSettings s;
+        s.setValue("cppCompilersList", QVariant::fromValue(mCompilersList));
+        emit compilersListChanged(mCompilersList);
+    }
+}
+
+void TabCpp::onCompilerListDoubleClicked(QListWidgetItem *item)
+{
+    QString initialText = item->text();
+    QString text = QInputDialog::getText(this, "Modify path"
+                                         ,"Modify path to compiler, please, or clear to remove path."
+                                         , QLineEdit::Normal
+                                         , item->text());
+
+    if (!text.size())           //  user removed item from list
+    {
+        mpCompilersList->takeItem(mpCompilersList->row(item)); // remove item from widget
+        for (int i = 0; i < mCompilersList.size(); ++i)        // remove item from list
```

Comment:
If you use method takeItem() then you are responsible for freeing memory 
You might also use method: void QListWidget::removeItemWidget(QListWidgetItem *item)
Then Qt guarantees that memory will be freed

Reply:
changed